cmd/otel-allocator/server/server.go

```diff
@@ -49,14 +48,6 @@ var (
 	}, []string{"path"})
 )
 
-var (
-	jsonConfig = jsoniter.Config{
-		EscapeHTML:                    false,
-		MarshalFloatWith6Digits:       true,
-		ObjectFieldMustBeSimpleString: true,
```

Comment:
Do you know how go-json behaves when it comes to these settings?

Reply:
it follows the same best practices as encoding/json. It doesn't cut any corners.

I assume test coverage would catch any regressions related to this change, is that an incorrect assumption?

Reply:
It should, I'm just a bit wary of Hyrum's Law. But looking through jsoniter's documentation, these seem to be performance optimizations, so we should be fine switching.

On that note, could you run the benchmark in https://github.com/open-telemetry/opentelemetry-operator/blob/main/cmd/otel-allocator/server/bench_test.go before and after this change?

Reply:
Yes, jsoniter was trying for higher performance by cutting corners, so if you have those flags off, there is no performance improvement from it. I have attached before/after/benchstat results below.

FWIW the collector project as a whole has moved over to goccy-json a little while back.